### ISSUE TITLE
[patch] Directive get ng-multiple and accept attributes too early

### ISSUE THREAD

**Issue by jeromemacias** [danialfarid/angular-file-upload#520]
In my application, configuration of `ng-multiple` and `accept` of the `ng-file-select` directive does not work.

To fix that, I need to modify the `ngFileSelect`directive definition:

```diff
@@ -177,7 +177,9 @@ angularFileUpload.directive('ngFileSelect', [ '$parse', '$timeout', '$compile',
 	restrict: 'AEC',
 	require:'?ngModel',
 	link: function(scope, elem, attr, ngModel) {
-		handleFileSelect(scope, elem, attr, ngModel, $parse, $timeout, $compile);
+		$timeout(function () {
+			handleFileSelect(scope, elem, attr, ngModel, $parse, $timeout, $compile);
+		}, 0);
 	}
 }}]);
```

Anyone had the same issue ?

**Comment by danialfarid** [comment]
Can you create a jsfiddle to show what problem you are having?

**Comment by danialfarid** [comment]
What version of the plugin are you using?

**Comment by jeromemacias** [comment]
I use version 2.2.2 and angular 1.3.11. I use the directive into another directive.
See https://github.com/marmelab/ng-admin/blob/f1020a6667ed7b2c5ee60e0576d3c983c8a2bd20/src/javascripts/ng-admin/Crud/field/maFileField.js for usage.

Sorry, it will be hard to create a jsfiddle. I test the demo localy with angular 1.3.11 and angular-file-upload
2.2.2 but it works on this simple example.

**Comment by hafezdivandari** [comment]
I have the similar issue and I think it's because the directive doesn't update the input element when attributes change, look at this fiddle: http://jsfiddle.net/z2seogbf/1/

**Comment by jeromemacias** [comment]
Thanks @hafezdivandari, you guided me to a solution.

I just set the multiple and accept scope attributes on pre-link, see https://github.com/marmelab/ng-admin/commit/55475a12d912477bccf44008093ee8be3945dd62?w=1

**Comment by danialfarid** [comment]
Since version 3.x the input file will be replaced and recompiled upon accept or multiple attributes model changes to reflect the new value for those attributes. So it should be fix since 3.0.0.
